sdk/azcore/policy_telemetry.go

```diff
@@ -11,31 +11,62 @@ import (
 	"fmt"
 	"os"
 	"runtime"
+	"strings"
 )
 
 // TelemetryOptions configures the telemetry policy's behavior.
 type TelemetryOptions struct {
 	// Value is a string prepended to each request's User-Agent and sent to the service.
 	// The service records the user-agent in logs for diagnostics and tracking of client requests.
 	Value string
+
+	// ApplicationID is an application-specific identification string used in telemetry.
+	// It has a maximum length of 24 characters and must not contain any spaces.
+	ApplicationID string
+
+	// Disabled will prevent the addition of any telemetry data to the User-Agent.
+	Disabled bool
```

Comment:
I think it would be better to have the code that creates the pipeline just exclude this policy entirely if disabled. I don't thin it's the "job" of the policy to ignore itself.

Reply:
Having the policy's configuration be self-contained means we only have to change it in one place (azcore) without making any changes to the code generator.  I don't see this being any different than setting the retry policy's MaxRetry count to zero to turn it off.

Reply:
Yes, that's true (about maxretries = 0) - OK, keep Disable as is.